includes/class-wc-shipping-zone.php

```diff
@@ -0,0 +1,404 @@
+		}
+
+		foreach ( $postcodes as $location ) {
+			$location_parts[] = $location->code;
+		}
+
+		// Fix display of encoded characters.
+		$location_parts = array_map( 'html_entity_decode', $location_parts );
+
+		if ( sizeof( $location_parts ) > $max ) {
+			$remaining = sizeof( $location_parts ) - $max;
+			return sprintf( _n( '%s and %d other region', '%s and %d other regions', $remaining, 'woocommerce' ), implode( ', ', array_splice( $location_parts, 0, $max ) ), $remaining );
+		} else {
+			return implode( ', ', $location_parts );
+		}
+	}
+
+	/**
+	 * Get shipping methods linked to this zone
+	 * @return array of objects
+	 */
+	public function get_shipping_methods() {
+		global $wpdb;
+
+        $raw_methods     = $wpdb->get_results( $wpdb->prepare( "SELECT method_id, method_order, instance_id FROM {$wpdb->prefix}woocommerce_shipping_zone_methods WHERE zone_id = %d order by method_order ASC;", $this->get_zone_id() ) );
+		$wc_shipping     = WC_Shipping::instance();
+		$allowed_classes = $wc_shipping->get_shipping_method_class_names();
+		$methods         = array();
+
+		foreach ( $raw_methods as $raw_method ) {
+			if ( in_array( $raw_method->method_id, array_keys( $allowed_classes ) ) ) {
+				$class_name = $allowed_classes[ $raw_method->method_id ];
+				if ( class_exists( $class_name ) ) {
+					$methods[ $raw_method->instance_id ]               = new $class_name( $raw_method->instance_id );
+					$methods[ $raw_method->instance_id ]->method_order = absint( $raw_method->method_order );
+					$methods[ $raw_method->instance_id ]->has_settings = $methods[ $raw_method->instance_id ]->has_settings();
+				}
+			}
+		}
+
+		return $methods;
+	}
+
+	/**
+	 * Location type detection
+	 * @param  object  $location
+	 * @return boolean
+	 */
+	private function location_is_continent( $location ) {
+		return 'continent' === $location->type;
+	}
+
+	/**
+	 * Location type detection
+	 * @param  object  $location
+	 * @return boolean
+	 */
+	private function location_is_country( $location ) {
+		return 'country' === $location->type;
+	}
+
+	/**
+	 * Location type detection
+	 * @param  object  $location
+	 * @return boolean
+	 */
+	private function location_is_state( $location ) {
+		return 'state' === $location->type;
+	}
+
+	/**
+	 * Location type detection
+	 * @param  object  $location
+	 * @return boolean
+	 */
+	private function location_is_postcode( $location ) {
+		return 'postcode' === $location->type;
+	}
+
+	/**
+	 * Set zone ID
+	 * @access private
+	 * @param int $set
+	 */
+    private function set_zone_id( $set ) {
+        $this->data['zone_id'] = absint( $set );
+    }
+
+	/**
+	 * Set zone name
+	 * @param string $set
+	 */
+    public function set_zone_name( $set ) {
+		$this->data['zone_name'] = wc_clean( $set );
+    }
+
+	/**
+	 * Set zone order
+	 * @param int $set
+	 */
+	public function set_zone_order( $set ) {
+        $this->data['zone_order'] = absint( $set );
+    }
+
+	/**
+     * Insert zone into the database
```

Comment:
Just wanted to point out the incorrect comment as I was looking at code here. :smile: